pkg/identity-manager/handlers.go

```diff
@@ -114,10 +120,67 @@ func (h *Handlers) ConfigureHandlers(api middleware.RoutableAPI) {
 	}
 
 	a.CookieAuth = func(token string) (interface{}, error) {
+		// For testing/dev environments only
+		if IdentityManagerFlags.SkipAuth {
+			log.Warn("Skipping authentication. This is not recommended in production environments.")
+			return "", nil
+		}
+		// Make a request to Oauth2Proxy to validate the cookie. Oauth2Proxy must be setup locally
+		proxyReq, err := http.NewRequest(http.MethodGet, IdentityManagerFlags.OAuth2ProxyAuthURL, nil)
+		if err != nil {
+			msg := "error creating forwarding request to oauth2proxy: %s"
+			log.Debugf(msg, err)
+			return nil, apiErrors.New(http.StatusUnauthorized, msg, err)
+		}
+
+		proxyReq.Header.Set("Cookie", token)
+		resp, err := http.DefaultClient.Do(proxyReq)
+		if err != nil {
+			msg := "error forwarding request to oauth2proxy: %s"
+			log.Debugf(msg, err)
+			return nil, apiErrors.New(http.StatusUnauthorized, msg, err)
+		}
+		if resp.StatusCode != http.StatusAccepted {
+			msg := "authentication failed with oauth2proxy: error code %v"
+			log.Debugf(msg, resp.StatusCode)
+			return nil, apiErrors.New(http.StatusUnauthorized, msg, resp.StatusCode)
+		}
+
+		// If authenticated, get subject
+		log.Debugf("Received Headers from oauth2proxy %s", resp.Header)
+		principal := resp.Header.Get(HTTPHeaderFwdEmail)
+		if principal == "" {
+			msg := "Authentication Failed: Missing %s header in response from oauth2proxy"
+			log.Debugf(msg, HTTPHeaderFwdEmail)
+			return nil, apiErrors.New(http.StatusUnauthorized, msg, HTTPHeaderFwdEmail)
+		}
+		// Valid Cookie return the subject
+		return principal, nil
+	}
+
+	a.BearerAuth = func(token string) (interface{}, error) {
+		// For testing/dev environments only
+		if IdentityManagerFlags.SkipAuth {
+			log.Warn("Skipping authentication. This is not recommended in production environments.")
+			return "", nil
+		}
+
+		parts := strings.Split(token, " ")
+		if len(parts) < 2 || strings.ToLower(parts[0]) != "bearer" {
+			msg := "invalid Authorization header, it must be of form 'Authorization: Bearer <token>'"
```

Comment:
Oops missed replying to this. You are right this is because go-swagger parses the `Authorization` header and provides the value to the bearerAuth function. But when returning the error message or during logging, I want to make sure the context is preserved.